web3/utils/datatypes.py

```diff
@@ -25,9 +25,10 @@ def __init__(cls, name, bases, namespace, **kargs):
         super().__init__(name, bases, namespace)
 
     def __new__(mcs, name, bases, namespace, normalizers=None):
+        all_bases = set(concat(base.__mro__ for base in bases))
         for key in namespace:
             verify_key_attr = verify_attr(name, key)
-            map(verify_key_attr, set(concat(base.__mro__ for base in bases)))
+            verify_key_attr(concat(base.__dict__ for base in all_bases))
```

Comment:
personal nitpick here.  I hate the implicit key iteration of dictionaries.  I personally have *never* internalized that dictionary iteration doesn't do `key/value` iteration and thus, have resorted to never using direct iteration on dictionaries, instead preferring `keys()/values()/items()` explicitely.

Can you change to `base.__dict__.keys()`.  Somewhat based on personal preference, but also it helps reduce cognitive load to understand this is pulling all the keys from all the dictionaries.

Reply:
If there is any agreement with this approach, I can add it to the tactical manual.

Reply:
I agree.  It makes it more clear.